packages/medusa/src/api-v2/admin/products/[id]/variants/[variant_id]/route.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  AuthenticatedMedusaRequest,
+  MedusaResponse,
+} from "../../../../../../types/routing"
+import {
+  deleteProductVariantsWorkflow,
+  updateProductVariantsWorkflow,
+} from "@medusajs/core-flows"
+
+import { UpdateProductVariantDTO } from "@medusajs/types"
+import { defaultAdminProductsVariantFields } from "../../../query-config"
+import { remoteQueryObjectFromString } from "@medusajs/utils"
+
+export const GET = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  const remoteQuery = req.scope.resolve("remoteQuery")
+
+  // TODO: Should we allow fetching a variant without knowing the product ID? In such case we'll need to change the route to /admin/products/variants/:id
+  const productId = req.params.id
+  const variantId = req.params.variant_id
+
+  const variables = { id: variantId, product_id: productId }
+
+  const queryObject = remoteQueryObjectFromString({
+    entryPoint: "product_variant",
+    variables,
+    fields: defaultAdminProductsVariantFields,
+  })
+
+  const [product_variant] = await remoteQuery(queryObject)
+  res.status(200).json({ product_variant })
+}
+
+export const POST = async (
+  req: AuthenticatedMedusaRequest<UpdateProductVariantDTO>,
+  res: MedusaResponse
+) => {
+  // TODO: Should we allow fetching a variant without knowing the product ID? In such case we'll need to change the route to /admin/products/variants/:id
+  const productId = req.params.id
+  const variantId = req.params.variant_id
+
+  const { result, errors } = await updateProductVariantsWorkflow(req.scope).run(
+    {
+      input: {
+        selector: { id: variantId, product_id: productId },
+        update: req.validatedBody,
+      },
+      throwOnError: false,
+    }
+  )
+
+  if (Array.isArray(errors) && errors[0]) {
+    throw errors[0].error
+  }
+
+  res.status(200).json({ product_variant: result[0] })
+}
+
+export const DELETE = async (
+  req: AuthenticatedMedusaRequest,
+  res: MedusaResponse
+) => {
+  // TODO: Should we allow fetching a variant without knowing the product ID? In such case we'll need to change the route to /admin/products/variants/:id
```

Comment:
With this specific endpoint path structure, I think it is fine to require the product ID. 

An alternative approach, if we want to support this, would be to introduce: `/admin/product-variants/:id`. That would make for pretty and consistent endpoonts.

Reply:
Same goes for all other comments on this.

Reply:
@olivermrbl a use-case I was thinking of is a case where you know the variant (eg. in the cart) but not the product, and you want to expand the relationship from variant -> product. Now if we have an endpoint like `/admin/product-variants/:id`, then do we even need the `product/:id/variant/:variant_id`? Anyway I am fine to leave things as they are for now, I am writing down all TODOs anyway and we can discuss this later on, just wanted to bring it up as a probably common scenario for nested models

Reply:
Makes sense.

> Now if we have an endpoint like /admin/product-variants/:id (which I think works well), then do we even need the product/:id/variant/:variant_id

If we introduce `/admin/product-variants/:id`, I would leave out `/product/:id/variant/:variant_id`. 

Sure, let's leave things as is and revisit later on.